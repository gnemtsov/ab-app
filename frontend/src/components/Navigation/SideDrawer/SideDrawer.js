import React from 'react';

import Logo from '../../Logo/Logo';
import NavigationItems from '../NavigationItems/NavigationItems';
import classes from './SideDrawer.css';
import Backdrop from '../../UI/Backdrop/Backdrop';
import Aux from '../../../hoc/Auxillary/Auxillary';

const sideDrawer = ( props ) => {
    let attachedClasses = [classes.SideDrawer, classes.Close];
    if (props.open) {
        attachedClasses = [classes.SideDrawer, classes.Open];
    }
    return (
        <Aux>
            <Backdrop show={props.open} clicked={props.closed}/>
            <div className={attachedClasses.join(' ')} onClick={props.closed}>
                <div className={classes.Logo}>
                    <Logo />
                </div>
                <nav>
                    <NavigationItems isAuthenticated={props.isAuth} />
                </nav>
            </div>
        </Aux>
    );
};

export default sideDrawer;