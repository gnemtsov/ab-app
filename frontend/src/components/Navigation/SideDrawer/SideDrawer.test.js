import React from 'react';

import { configure, render, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import SideDrawer from './SideDrawer';

configure({ adapter: new Adapter() });

describe('SideDrawer', () => {
	it('Render closed, not authenticated', () => {
		const wrapper = shallow (
	        <SideDrawer 
				open={false}
				isAuth={false}
				/>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Render closed, authenticated', () => {
		const wrapper = shallow (
	        <SideDrawer 
				open={false}
				isAuth={true}
				/>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Render opened, not authenticated', () => {
		const wrapper = shallow (
	        <SideDrawer 
				open={true}
				isAuth={false}
				/>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
	
	it('Render opened, authenticated', () => {
		const wrapper = shallow (
	        <SideDrawer 
				open={true}
				isAuth={true}
				/>
		);
		
		expect(wrapper).toMatchSnapshot();
	});
});
