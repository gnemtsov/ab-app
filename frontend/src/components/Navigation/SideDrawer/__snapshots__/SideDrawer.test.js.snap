// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideDrawer Render closed, authenticated 1`] = `
<Fragment>
  <backdrop
    show={false}
  />
  <div
    className="SideDrawer Close"
  >
    <img
      alt="AB-APP logo"
      className="Logo"
      src="test-pic-stub"
    />
    <nav>
      <navigationItems
        isAuthenticated={true}
      />
    </nav>
  </div>
</Fragment>
`;

exports[`SideDrawer Render closed, not authenticated 1`] = `
<Fragment>
  <backdrop
    show={false}
  />
  <div
    className="SideDrawer Close"
  >
    <img
      alt="AB-APP logo"
      className="Logo"
      src="test-pic-stub"
    />
    <nav>
      <navigationItems
        isAuthenticated={false}
      />
    </nav>
  </div>
</Fragment>
`;

exports[`SideDrawer Render opened, authenticated 1`] = `
<Fragment>
  <backdrop
    show={true}
  />
  <div
    className="SideDrawer Open"
  >
    <img
      alt="AB-APP logo"
      className="Logo"
      src="test-pic-stub"
    />
    <nav>
      <navigationItems
        isAuthenticated={true}
      />
    </nav>
  </div>
</Fragment>
`;

exports[`SideDrawer Render opened, not authenticated 1`] = `
<Fragment>
  <backdrop
    show={true}
  />
  <div
    className="SideDrawer Open"
  >
    <img
      alt="AB-APP logo"
      className="Logo"
      src="test-pic-stub"
    />
    <nav>
      <navigationItems
        isAuthenticated={false}
      />
    </nav>
  </div>
</Fragment>
`;
